### PR TITLE
failover: add supervised failover commands

### LINKED ISSUE #822: Support the manual master switch using the failover coordinator
Something like the following.

```sh
$ tt failover switch i-002 [--timeout 10] [-d]
<..here command updates are shown if -d is omitted..>
```

See https://github.com/tarantool/tarantool-ee/pull/759 for details.

### PR DESCRIPTION
**Pull request by 0x501D**
@TarantoolBot document
Title: `tt cluster failover commands`

This patch adds new commands:
* tt cluster failover switch - switch master instance
* tt cluster failover switch-status - show master switching status

This commands works only with etcd based failover configuration.

Closes #822

### CODE REVIEW

File: test/integration/cluster/test_cluster_failover.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+timeout: 42
+"""
+
+
+def test_cluster_failover_switch_etcd(tt_cmd, tmpdir_with_cfg, etcd):
+    tmpdir = tmpdir_with_cfg
+
+    conn = etcd.conn()
+    switch_cmd = [tt_cmd, "cluster", "failover", "switch",
+                  f"{etcd.endpoint}/prefix", "some_instance"]
```
Comment (psergee) [inline test/integration/cluster/test_cluster_failover.py:20]: Should it work if there is no `some_instance` instance in cluster?

File: cli/cmd/cluster.go [hunk 2]
```diff
@@ -169,6 +210,56 @@ func newClusterReplicasetCmd() *cobra.Command {
...
+		Use:     "failover",
+		Short:   "Manage supervised failover",
+		Aliases: []string{"fo"},
+	}
+
+	switchCmd := &cobra.Command{
+		Use:                   "switch <URI> <INSTANCE_NAME> [flags]",
+		DisableFlagsInUseLine: true,
+		Short:                 "Switch master instance",
+		Long:                  "Switch master instance\n\n" + failoverUriHelp,
```
Comment (psergee) [inline cli/cmd/cluster.go:224]: Can you please add usage example?

**Review by psergee — COMMENTED** [review]
Thank you for the patch!

File: test/integration/cluster/test_cluster_failover.py (hunk 1 above)
Comment (0x501D) [reply test/integration/cluster/test_cluster_failover.py:20]: Yes. Return will be like:
```yml
new_master: none
command: switch
status: failed
taken_by:
  active: true
  term: 10
  uuid: de36e15a-d2a5-4015-b378-a8eb262c8ff6
  hostname: void-tt
  pid: 18209
timeout: 30
reason: Unknown instance name
```

File: cli/cluster/cmd/failover.go [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+	if err != nil {
+		return err
+	}
+
+	if len(result.Kvs) != 1 {
+		return fmt.Errorf("task with id `%s` is not found", switchCtx.TaskID)
+	}
+
+	fmt.Print(string(result.Kvs[0].Value))
```
[the remark this answers is not in the document]
Comment (0x501D) [reply cli/cluster/cmd/failover.go:198]: Yes. The value has new-line char already.

File: cli/cluster/cmd/failover.go [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+			case <-ctx.Done():
+				fmt.Println("Timeout for command execution reached.")
+				return nil
+			case ev := <-outputChan:
+				result := switchCmdResult{}
+				err = yaml.Unmarshal(ev.Kv.Value, &result)
+				if err != nil {
+					return err
+				}
+				fmt.Printf("%s", ev.Kv.Value)
```
[the remark this answers is not in the document]
Comment (0x501D) [reply cli/cluster/cmd/failover.go:145]: Additional context not needed:

output example:
```yml
$ tt cluster failover switch http://localhost:2379/tcs instance2 --wait 
command: switch
new_master: instance2
timeout: 30
---
new_master: instance2
command: switch
status: taken
timeout: 30
taken_by:
  active: true
  term: 10
  uuid: de36e15a-d2a5-4015-b378-a8eb262c8ff6
  hostname: void-tt
  pid: 18209
...
---
old_master: instance1
taken_by:
  active: true
  term: 10
  uuid: de36e15a-d2a5-4015-b378-a8eb262c8ff6
  hostname: void-tt
  pid: 18209
timeout: 30
new_master: instance2
command: switch
status: success
result: Switched a master
replicaset_name: replicaset1
...
```

File: cli/cmd/cluster.go (hunk 2 above)
Comment (0x501D) [reply cli/cmd/cluster.go:224]: Added

File: cli/cluster/cmd/failover.go [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+	etcd, err := libcluster.ConnectEtcd(opts)
+	if err != nil {
+		return fmt.Errorf("unable to connect to etcd: %w", err)
+	}
+	defer etcd.Close()
+
+	key := uriOpts.Prefix + failoverPath + switchCtx.TaskID
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultEtcdTimeout)
+	result, err := etcd.Get(ctx, key, clientv3.WithLimit(1))
```
[the remark this answers is not in the document]
Comment (0x501D) [reply cli/cluster/cmd/failover.go:195]: Added

File: cli/cluster/cmd/failover.go [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+					case mvccpb.PUT:
+						outputChan <- ev
+					}
+				}
+			}
+			cancel()
+		}()
+
+		ctx_put, cancel := context.WithTimeout(context.Background(), defaultEtcdTimeout)
+		_, err = etcd.Put(ctx_put, key, string(yamlCmd))
```
[the remark this answers is not in the document]
Comment (0x501D) [reply cli/cluster/cmd/failover.go:132]: Added

File: cli/cluster/cmd/failover.go [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+				return nil
+			case ev := <-outputChan:
+				result := switchCmdResult{}
+				err = yaml.Unmarshal(ev.Kv.Value, &result)
+				if err != nil {
+					return err
+				}
+				fmt.Printf("%s", ev.Kv.Value)
+				if len(result.Result) > 0 {
+					return nil
```
[the remark this answers is not in the document]
Comment (0x501D) [reply cli/cluster/cmd/failover.go:155]: Done

File: cli/cluster/cmd/failover.go [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+		ctx_put, cancel := context.WithTimeout(context.Background(), defaultEtcdTimeout)
+		_, err = etcd.Put(ctx_put, key, string(yamlCmd))
+		cancel()
+
+		for {
+			select {
+			case <-ctx.Done():
+				fmt.Println("Timeout for command execution reached.")
+				return nil
+			case ev := <-outputChan:
```
[the remark this answers is not in the document]
Comment (0x501D) [reply cli/cluster/cmd/failover.go]: Fixed

File: cli/cluster/cmd/failover.go [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+		}()
+
+		ctx_put, cancel := context.WithTimeout(context.Background(), defaultEtcdTimeout)
+		_, err = etcd.Put(ctx_put, key, string(yamlCmd))
+		cancel()
+
+		for {
+			select {
+			case <-ctx.Done():
+				fmt.Println("Timeout for command execution reached.")
```
[the remark this answers is not in the document]
Comment (0x501D) [reply cli/cluster/cmd/failover.go]: Fixed

**Pull request merged**